### PR TITLE
Live: expose HTTP push endpoint that will read influx line protocol and publish to websocket

### PR DESCRIPTION
**Pull request by FZambia**
**What this PR does / why we need it**:

If Live feature on we can start receiving metrics from Telegraf configured like this:

```toml
[agent]
  interval = "1s"
  flush_interval = "1s"

[[inputs.system]]
  fielddrop = ["uptime_format", "uptime"]
  [inputs.system.tags]
    mylabel = "boom"

[[inputs.cpu]]
  totalcpu = true

[[outputs.http]]
  url = "http://localhost:3000/api/live/push/mypath"
  method = "POST"
  data_format = "influx"
  content_encoding = "identity"

  [outputs.http.headers]
    Authorization = "Bearer <Grafana API Token>"
```

Input will be converted to `data.Frame`s and published to channels.

To see the data in the browser, use the `-- Grafana --` datasource with "Live Measurements" and enter the channel path.  The path will relate to your request and measurement name

image

### CODE REVIEW

File: pkg/api/api.go [hunk 1]
```diff
@@ -399,6 +399,10 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// short urls
 		apiRoute.Post("/short-urls", bind(dtos.CreateShortURLCmd{}), routing.Wrap(hs.createShortURL))
+
+		if hs.Live.IsEnabled() {
+			apiRoute.Post("/live/telemetry/*", hs.LiveTelemetry.Handle)
```
Comment (FZambia) [inline pkg/api/api.go:404]: not very happy to pass channel path as part of API url here - looks like alternatives are custom header or url param.

Comment (ryantxu) [reply pkg/api/api.go:408]: do we want "telemetry" as an additional scope?  if so, does this path conflict?

Comment (FZambia) [reply pkg/api/api.go:408]: At this moment everything published to this endpoint goes to `telemetry/{PATH}/{METRIC_NAME}` where `{PATH}` is `*` from route handler path `/live/telemetry/*`.

So:
* POST to `/live/telemetry/test` will result into publish into `telemetry/test/cpu`, `telemetry/test/system` 
* POST to `/live/telemetry/test/2` will result into publish into `telemetry/test/2/cpu`, `telemetry/test/2/system`

But it's not obvious to me whether this way of channel construction is ok or not.

Comment (ryantxu) [reply pkg/api/api.go:408]: I thin we should require that the POST 'path' component does not contain a slash -- ie `POST to /live/telemetry/test/2` should be an error, but POST to /live/telemetry/test-2 is OK

Comment (ryantxu) [reply pkg/api/api.go:408]: maybe:
```
                if hs.Live.IsEnabled() {
			apiRoute.Post("/live/push/:id", hs.LiveTelemetry.Handle)
		}
		```

where id has to be a simple string?

**Review by ryantxu — APPROVED** [review]
Looks great!

**Pull request merged**